Sources/NIOCore/ChannelPipeline.swift

```diff
@@ -2089,3 +2089,171 @@ extension ChannelPipeline: CustomDebugStringConvertible {
         return handlers
     }
 }
+
+extension ChannelPipeline {
+    private enum BufferingDirection: Equatable {
+        case inbound
+        case outbound
+    }
+
+    /// Retrieve the total number of bytes buffered for outbound.
+    public func outboundBufferedBytes() -> EventLoopFuture<Int> {
+        let future: EventLoopFuture<Int>
+
+        if self.eventLoop.inEventLoop {
+            future = self.eventLoop.makeSucceededFuture(countAllBufferedBytes(direction: .outbound))
+        } else {
+            future = self.eventLoop.submit {
+                self.countAllBufferedBytes(direction: .outbound)
+            }
+        }
+
+        return future
+    }
+
+    /// Retrieve the number of outbound bytes buffered in the `ChannelHandler` associated with the given `ChannelHandlerContext`.
+    ///
+    /// - Parameters:
+    ///     - in: the `ChannelHandlerContext` from which the outbound buffered bytes of the `ChannelHandler` will be retrieved.
+    ///
+    /// - Returns: The `EventLoopFuture` which will be notified once the number of outbound bytes buffered in the `ChannelHandler`
+    ///            referenced by the `ChannelHandlerContext` parameter `in` is collected.
+    ///            If the `ChannelHandler` in the `ChannelHandlerContext` does not conform to
+    ///            `NIOOutboundByteBufferingChannelHandler`, the future will contain`nil`.
+    public func outboundBufferedBytes(in context: ChannelHandlerContext) -> EventLoopFuture<Int?> {
```

Comment:
Looking at this API I'm not really sure it's right. Users can't safely hold a `ChannelHandlerContext` from _off_ the `EventLoop` (they aren't `Sendable`), so I don't think this API makes sense here. It only really makes sense on the `syncOperations` view.

We could write alternative APIs here that _do_ work from off the EL, but I'm not all that sure those are useful anyway.

Reply:
Thanks for catching this! Indeed, we cannot ensure that `ChannelHandlerContext` is safe unless user uses

```swift
try channel.pipeline.context(name: inboundChannelHandlerName).flatMap { context in
      channel.pipeline.inboundBufferedBytes(in: context)
}
```